core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -378,8 +426,10 @@ class TransactionsTest extends KafkaServerTestHarness {
     val records = new ArrayBuffer[ConsumerRecord[Array[Byte], Array[Byte]]]()
     TestUtils.waitUntilTrue(() => {
```

Comment:
This method seems very similar to `TestUtils.consumeRecords`. Maybe we can consolidate.